### DISCUSSION TITLE
Possible to apply packEnclose to force-tree-graphs?

### DISCUSSION THREAD

**Discussion by vonKristoff** [d3/d3#3960] [Q&A] [answered]
I was looking at this example https://observablehq.com/@d3/force-directed-tree

Because the nodes are always in simulation - is there a way to redraw a packEnclosed [Source](https://github.com/d3/d3-hierarchy/blob/main/src/pack/enclose.js) circle over a selected group?

My fork is here https://observablehq.com/d/b70c9953ba85e3f7

**Comment by Fil** [comment, marked as answer]
The trick is to recompute the enclosure each time you redraw:

```js
    // APPLY Packed POSITION
    const enclosure = d3.packEnclose(w.get(true).map((d) => ({ ...d, r: 1 })));
    group
	.attr('cx', enclosure.x)
	.attr('cy', enclosure.y)
	.attr('r', enclosure.r);
```

**Reply by vonKristoff** [reply]
Amazing @Fil TY again. Appreciated. 
Also I am beginning to see how this works now
